### PR TITLE
[dev] add yapf to pre-commit

### PR DESCRIPTION
**Pull request by dcramer**
Based on facebook styles, with a few minor tweaks:

https://github.com/google/yapf/blob/9f168a120bac07596c10454d4465e63ce1dd7182/yapf/yapflib/style.py#L200-L270

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -33,3 +33,4 @@ example/db.sqlite
 /test_cli/
 Gemfile.lock
 .idea/
+.vscode/tags
```
Comment (dcramer) [inline .gitignore]: will backout this unrelated change

**Review by mitsuhiko — APPROVED** [review]

File: config/hooks/pre-commit [hunk 2]
```diff
@@ -9,18 +9,19 @@ import json
...
-from sentry.lint.engine import check_files, get_js_files
+from sentry.lint.engine import check_files, get_js_files, get_python_files
 
 text_type = type(u'')
 
 # git usurbs your bin path for hooks and will always run system python
 if 'VIRTUAL_ENV' in os.environ:
-    site_packages = glob(
-        '%s/lib/*/site-packages' % os.environ['VIRTUAL_ENV'])[0]
+    site_packages = glob('%s/lib/*/site-packages' % os.environ['VIRTUAL_ENV']
```
Comment (dcramer) [inline config/hooks/pre-commit]: im not happy with this, but i prefer the dedent option in 90% of the cases

**Comment by tkaemming** [comment]
Is there a way to disable this dictionary reformatting?

```diff
diff --git a/src/sentry/digests/notifications.py b/src/sentry/digests/notifications.py
index 4bfe76a..0fcce50 100644
--- a/src/sentry/digests/notifications.py
+++ b/src/sentry/digests/notifications.py
@@ -63,13 +71,25 @@ def fetch_state(project, records):
     start = records[-1].datetime
     end = records[0].datetime
 
-    groups = Group.objects.in_bulk(record.value.event.group_id for record in records)
+    groups = Group.objects.in_bulk(
+        record.value.event.group_id for record in records
+    )
     return {
-        'project': project,
-        'groups': groups,
-        'rules': Rule.objects.in_bulk(itertools.chain.from_iterable(record.value.rules for record in records)),
-        'event_counts': tsdb.get_sums(tsdb.models.group, groups.keys(), start, end),
-        'user_counts': tsdb.get_distinct_counts_totals(tsdb.models.users_affected_by_group, groups.keys(), start, end),
+        'project':
+        project,
+        'groups':
+        groups,
+        'rules':
+        Rule.objects.in_bulk(
+            itertools.chain.
+            from_iterable(record.value.rules for record in records)
+        ),
+        'event_counts':
+        tsdb.get_sums(tsdb.models.group, groups.keys(), start, end),
+        'user_counts':
+        tsdb.get_distinct_counts_totals(
+            tsdb.models.users_affected_by_group, groups.keys(), start, end
+        ),
     }
```

I also personally don't like this space-after-single-member tuple formatting but don't really care that much:

```diff
diff --git a/src/sentry/digests/backends/redis.py b/src/sentry/digests/backends/redis.py
index 7d44d23..ad2073f 100644
--- a/src/sentry/digests/backends/redis.py
+++ b/src/sentry/digests/backends/redis.py
@@ -428,7 +452,10 @@ class RedisBackend(Backend):
             if len(entries) < fetch_size:
                 break
         else:
-            raise RuntimeError('loop exceeded maximum iterations (%s)' % (maximum_iterations,))
+            raise RuntimeError(
+                'loop exceeded maximum iterations (%s)' %
+                (maximum_iterations, )
+            )
 
     def maintenance(self, deadline, chunk=1000):
         # TODO: Ideally, this would also return the number of items that were
```

**Comment by dcramer** [comment]
deferring this for now
